draft-ietf-masque-h3-datagram.md

```diff
@@ -466,11 +509,38 @@ endpoint receives a CLOSE_DATAGRAM_CONTEXT capsule that violates one or more of
 these requirements, the endpoint MUST abruptly terminate the corresponding
 stream with a stream error of type H3_GENERAL_PROTOCOL_ERROR.
 
-All CLOSE_DATAGRAM_CONTEXT capsules MUST contain a CLOSE_CODE context
-extension, see {{close-code}}. If an endpoint receives a CLOSE_DATAGRAM_CONTEXT
-capsule without a CLOSE_CODE context extension, the endpoint MUST abruptly
-terminate the corresponding stream with a stream error of type
-H3_GENERAL_PROTOCOL_ERROR.
+
+#### Close Codes {#close-codes}
+
+This specification defines the close codes below. Their numeric values are in
+{{iana-close-codes}}.
+
+NO_ERROR:
+
+: This indicates that the registration was closed without any additional
+information.
+
+UNKNOWN_FORMAT:
+
+: This indicates that the sender does not know how to interpret the datagram
+format type associated with this context. The endpoint that had originally
+registered this context MUST NOT try to register another context with the same
+datagram format type on this stream.
+
+DENIED:
+
+: This indicates that the sender has rejected the context registration based on
+its local policy. The endpoint that had originally registered this context MUST
+NOT try to register another context with the same datagram format type and
+datagram format data on this stream.
+
```

Comment:
Another reason for rejecting a registration, for example, would be when receiving a Format Type of IP_PACKET on a CONNECT-UDP stream. Perhaps we can defined another code for that obviously broken case?

Reply:
Agreed, though the reaction to the code would be the same (MUST NOT use this type again) so I don't think we need a separate code

Reply:
yeah renaming DENIED to REJECTED and just stating policy or unsupported as examples could work. It's not a blocker to this PR though